### PR TITLE
LegendasTV: fix caching titles for wrong season

### PR DESCRIPTION
**Pull request by fernandog**
It should only cache titles that matches give season when episode type
@ratoaq2 if ok, I will commit upstream to the respective PR

### CODE REVIEW

File: medusa/subtitle_providers/legendastv.py [hunk 1]
```diff
@@ -258,6 +258,12 @@ def search_titles(self, title, season):
                     else:
                         logger.warning('No season detected for title %d (%s)', title_id, name)
 
+                # discard title mismatches on season
```
Comment (ratoaq2) [inline medusa/subtitle_providers/legendastv.py]: Now you have parts of the mismatches discard logic in `search_titles` and parts in `query`.

Comment (fernandog) [inline medusa/subtitle_providers/legendastv.py]: Title is associated to the season, so it must be here.

**Comment by fernandog** [comment]
@ratoaq2 I rewrote. I had to rename `title` var because you override the one from argument when you `title = {'type': type_map[source['tipo']]}` - same with new param `year` I added and year from result

**Comment by p0psicles** [comment]
@ratoaq2, as far as I can review it, it looks good. Shall we approve?

**Pull request merged**